### DISCUSSION TITLE
monorepo needs to deploy 2 different projects on different subdomains

### DISCUSSION THREAD

**Discussion by ottonascarella** [vercel/community#7404] [Help]
### Summary

Hiya,

we currently have a project, let's call it `A`, that is running on domain `a.stuff.com`.

We'd like to have another project on the same monorepo, but would like it to be deployed to a different subdomain... `b.stuff.com`. Effectively it's a different project, but as project `B` shares loads of modules with project `A`, we want it to be in the same monorepo.

Is there a way to configure that given that `vercel.json` has only one `outputDirectory`?

### Steps to Reproduce

It's not a bug.

**Comment by pawlean** [comment]
Hi @ottonascarella!

I believe you can do this. Your monorepo would look something like:
```
/monorepo
  /projectA
    - vercel.json
    - package.json
    - ...
  /projectB
    - vercel.json
    - package.json
    - ...
  - package.json (for monorepo)
  - ...
```

For each `vercel.json` file, you'd have something like:

```
{
  "builds": [
    { "src": "projectA/**/*", "use": "@vercel/node" }
  ],
  "routes": [
    { "src": "/(.*)", "dest": "projectA/$1" }
  ],
  "outputDirectory": "projectA/build"
}
```

You can then create separate projects in the Vercel dashboard, one for each subdomain where you set the root directory to projectA and define the build and output settings according to `projectA/vercel.json`.

Sharing some helpful links:

- https://vercel.com/guides/monorepos
- https://github.com/vercel/examples/tree/main/solutions/monorepo
- https://vercel.com/docs/project-configuration

Let us know how you get on!

[a comment by an automated account is not shown]
